libtcmu_config.c

```diff
@@ -356,11 +377,115 @@ static void tcmu_parse_options(struct tcmu_config *cfg, char *buf, int len)
 
 }
 
-int tcmu_load_config(struct tcmu_config *cfg, const char *path)
+int tcmu_reload_config(struct tcmu_config *cfg)
 {
 	char buf[TCMU_MAX_CFG_FILE_SIZE];
 	int fd, len;
 
+	fd = open(cfg->path, O_RDONLY);
+	if (fd < 0) {
+		tcmu_err("Failed to open file '%s', %m\n", cfg->path);
+		return -1;
+	}
+
+	len = tcmu_read_config(fd, buf, TCMU_MAX_CFG_FILE_SIZE);
+	close(fd);
+	if (len < 0) {
+		tcmu_err("Failed to read file '%s'\n", cfg->path);
+		return -1;
+	}
+
+	buf[len] = '\0';
+
+	tcmu_parse_options(cfg, buf, len);
+
+	return 0;
+}
+
+static void dyn_config_cleanup(void *arg)
+{
+	struct tcmu_config *cfg = arg;
+	struct tcmu_conf_option *opt;
+
+	darray_foreach(opt, tcmu_options) {
+		if (opt->type == TCMU_OPT_STR)
+			free(opt->opt_str);
+	}
+
+	darray_free(tcmu_options);
+	free(cfg->path);
+	free(cfg);
+}
```

Comment:
You could probably put this in a function like destroy_config or config_cleanup and call it from here and the case where we are not doing dynamic config support.